### DISCUSSION TITLE
Is there a way to ignore specific LSP warnings from within the source?

### DISCUSSION THREAD

**Discussion by aral** [helix-editor/helix#3182] [Q&A]
I have a JavaScript warning `“Property '_version' may not exist on type 'Sade'. Did you mean 'version'?”`

This is because I’m being naughty and knowingly overwriting a private method.

Is there some way I can signal to Helix that I know I’m being naughty and it’s ok (this one time?) ;)

e.g., to signal that an LSP warning is to be expected for the next line:

```js
// Overwrite the built-in _version() method called by the Sade version
// command to format the output and to also manually exit the process
// otherwise it will hang due to the open broadcast channel between the
// loader process and the main process.

// @lsp-expect-warning Property '_version' may not exist on type 'Sade'
this.commands._version = function () { 
  console.log(`😸️ Kitten version ${this.ver} – Production Mode`)
  process.exit(0)
}
```

**Comment by sudormrfbin** [comment]
This is usually handled by the LSP itself through some kind of comment or annotation in the code, like `#[clippy::allow_unused_imports]` in Rust or `# pylint: disable=locally-disabled` in Python. Maybe there's something similar for the JS language server?

**Reply by aral** [reply]
Ah, yeah, there are // @ts-… comments but I wasn’t able to get them to work hence the question. I’ll take another look. Maybe I was holding it wrong or something :)
